api/api_public_transaction_pool.go

```diff
@@ -149,6 +150,55 @@ func (s *PublicTransactionPoolAPI) GetTransactionByHash(ctx context.Context, has
 	return nil
 }
 
+// GetDecodedAnchoringTransactionByHash returns the decoded anchoring data of anchoring transaction for the given hash
+func (s *PublicTransactionPoolAPI) GetDecodedAnchoringTransactionByHash(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
+	var tx *types.Transaction
+
+	// Try to return an already finalized transaction
+	if tx, _, _, _ = s.b.ChainDB().ReadTxAndLookupInfo(hash); tx != nil {
+		goto decode
+	}
+
+	// No finalized transaction, try to retrieve it from the pool
+	if tx = s.b.GetPoolTransaction(hash); tx != nil {
+		goto decode
+	}
+	return nil, errors.New("can't find the transaction")
+
+decode:
```

Comment:
Why don't you use `types.DecodeAnchoringData`?

Reply:
Thanks. @aidan-kwon I reflected. :)